velero-plugins/pod/restore.go

```diff
@@ -194,42 +196,64 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	if input.Restore.Spec.NamespaceMapping[destNamespace] != "" {
 		destNamespace = input.Restore.Spec.NamespaceMapping[destNamespace]
 	}
-	secretList, err := client.Secrets(destNamespace).List(context.Background(), metav1.ListOptions{})
+
+	var secretList *corev1API.SecretList
+	nameSpace, err := client.Namespaces().Get(context.Background(), destNamespace, metav1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
-	nameSpace, err := client.Namespaces().Get(context.Background(), destNamespace, metav1.GetOptions{})
+
+	// We check for the existence of OpenShift Image Registry replicas to determine whether ImageRegsitry Cluster capabilities are enabled 
+	// Additionally we also need to check OCP version 
+	// Based on the above 2 things we determine whether to skip waiting for docker secret i.e.  if image resgistry is not enabled and OCP cluster is below 4.15
+
+	ocpRegistryHasReplicas, err := openshift.ImageRegistryHasReplicas()
+	if err != nil {
+		return nil, err
+	}
+
+	ocpVersion, err := openshift.GetClusterVersion()
+	if err != nil {
+		return nil, err
+	}
+
+	_, minorVersion, _, err := common.ParseOCPVersion(ocpVersion.Status.Desired.Version)
 	if err != nil {
 		return nil, err
 	}
-	for true {
-		flag := 0
-		for _, secret := range secretList.Items {
-			if strings.HasPrefix(secret.Name, "default-dockercfg-") {
-				p.Log.Info(fmt.Sprintf("[pod-restore] Found new dockercfg secret: %v", secret))
-				flag = 1
+
+	minorVersionInt, _ := strconv.Atoi(minorVersion)
+
+	if ocpRegistryHasReplicas && minorVersionInt >= 15 {
```

Comment:
nit: Should probably replace `minorVersionInt >= 15` with `majorVersionInt == 4 && minorVersionInt >= 15 || majorVersion > 4` -- otherwise we may get a mysterious break when we move to ocp 5.0 in the future and nobody remembers making this fix.

Reply:
(will require processing majorVersion in line 220 to do the above)

Reply:
Also, rather than having to do all of this lookup/calculation for every pod, maybe we do it once and store the results as a bool in the RestorePlugin struct? See https://github.com/openshift/openshift-velero-plugin/blob/master/velero-plugins/serviceaccount/backup.go#L23 for an example of this.

Reply:
Done, updated the PR.